src/main/java/com/github/mmolimar/kafka/connect/fs/policy/AbstractPolicy.java

```diff
@@ -197,17 +198,34 @@ public FileReader offer(FileMetadata metadata, OffsetStorageReader offsetStorage
                 .filter(fs -> metadata.getPath().startsWith(fs.getWorkingDirectory().toString()))
                 .findFirst()
                 .orElse(null);
+
+        Supplier<FileReader> makeReader = () -> ReflectionUtils.makeReader(
+                (Class<? extends FileReader>) conf.getClass(FsSourceTaskConfig.FILE_READER_CLASS),
+                current, new Path(metadata.getPath()), conf.originals()
+        );
         try {
-            FileReader reader = ReflectionUtils.makeReader(
-                    (Class<? extends FileReader>) conf.getClass(FsSourceTaskConfig.FILE_READER_CLASS),
-                    current, new Path(metadata.getPath()), conf.originals());
             Map<String, Object> partition = Collections.singletonMap("path", metadata.getPath());
             Map<String, Object> offset = offsetStorageReader.offset(partition);
             if (offset != null && offset.get("offset") != null) {
+                Object offsetObject = offset.get("offset");
+                Object fileSizeBytesObject = offset.get("fileSizeBytes");
+
+                // Only new versions of kafka-connect-fs store the file size bytes
+                // If we have the byte offset, we can skip reading the entire file if the file size has not changed
+                if(fileSizeBytesObject != null) {
+                    Long byteOffset = (Long)fileSizeBytesObject;
+                    if (metadata.getLen() == byteOffset){
+                        log.info("File {} has byte length and byte offset of: {}, skipping reading the file as it is unchanged since the last execution", metadata.getPath(), byteOffset);
+                        return null;
```

Comment:
I'd prefer returning here a "fake" file reader object implementing the ``FileReader`` interface with no elements.

Reply:
There are a lot of methods to implement and it complicates the code a lot for a simple operation (is this thing present?). 

Is using `java.util.Optional` ok instead? This connector only supports java8 or higher so it seems reasonable to use.